h/activity/query.py

```diff
@@ -80,9 +81,15 @@ def check_url(request, query, unparse=parser.unparse):
 
     if _single_entry(query, 'group'):
         pubid = query.pop('group')
-        redirect = request.route_path('activity.group_search',
-                                      pubid=pubid,
-                                      _query={'q': unparse(query)})
+
+        try:
+            group = request.db.query(Group).filter_by(pubid=pubid).one()
+        except exc.NoResultFound:
```

Comment:
It's a small thing, but you'd save yourself an import here and a couple of lines if you used `.one_or_none()`.